File: weknot_android/app/src/main/java/com/example/weknot_android/room/dao/UserDao.java
package com.example.weknot_android.room.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.weknot_android.room.entity.user.User;

@Dao
public interface UserDao {

    @Insert
    void insert(User user);

    @Update
    void update(User user);

    @Delete
    void delete(User user);

    @Query("DELETE FROM user_table")
    void deleteAll();

    @Query("SELECT * FROM user_table WHERE id=:id")
    LiveData<User> getUser(String id);
}

File: weknot_android/app/src/main/java/com/example/weknot_android/room/dao/UserDao.kt
package com.example.weknot_android.room.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query

import com.example.weknot_android.base.BaseDao
import com.example.weknot_android.room.entity.user.User
import io.reactivex.Single

@Dao
interface UserDao : BaseDao<User> {

    @Query("DELETE FROM user_table")
    fun deleteAll()

    @Query("SELECT * FROM user_table WHERE id=:id")
    fun getUser(id: String): Single<User>?
}
